datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/system/cdc/debezium/ConfigureDebeziumConnectorStep.java

```diff
@@ -0,0 +1,278 @@
+
+  private final DebeziumConfiguration debeziumConfig;
+  private final EbeanConfiguration ebeanConfig;
+  private final KafkaConfiguration kafkaConfig;
+  private final KafkaProperties kafkaProperties;
+
+  public ConfigureDebeziumConnectorStep(
+      OperationContext opContext,
+      DebeziumConfiguration debeziumConfig,
+      EbeanConfiguration ebeanConfig,
+      KafkaConfiguration kafkaConfig,
+      KafkaProperties kafkaProperties) {
+    this.opContext = opContext;
+    this.debeziumConfig = debeziumConfig;
+    this.ebeanConfig = ebeanConfig;
+    this.kafkaConfig = kafkaConfig;
+    this.kafkaProperties = kafkaProperties;
+  }
+
+  @Override
+  public String id() {
+    return "ConfigureDebeziumConnectorStep";
+  }
+
+  @Override
+  public Function<UpgradeContext, UpgradeStepResult> executable() {
+    return (context) -> {
+      log.info("Configuring Debezium connector for CDC processing...");
+
+      try {
+        String connectUrl = debeziumConfig.getUrl();
+        if (connectUrl == null || connectUrl.isEmpty()) {
+          log.error(
+              "Kafka Connect URL not configured in CDC configuration - cannot configure connector");
+          return new DefaultUpgradeStepResult(this.id(), DataHubUpgradeState.FAILED);
+        }
+
+        Map<String, Object> connectorConfig = buildConnectorConfiguration();
+        if (connectorConfig.isEmpty()) {
+          log.error("Failed to build connector configuration");
+          return new DefaultUpgradeStepResult(this.id(), DataHubUpgradeState.FAILED);
+        }
+        HttpClient httpClient = createHttpClient();
+
+        String connectorName = debeziumConfig.getName();
+        if (connectorExists(httpClient, connectUrl, connectorName)) {
+          return updateConnector(httpClient, connectUrl, connectorName, connectorConfig);
+        } else {
+          return createConnector(httpClient, connectUrl, connectorName, connectorConfig);
+        }
+
+      } catch (Exception e) {
+        log.error("Failed to configure Debezium connector: {}", e.getMessage(), e);
+        return new DefaultUpgradeStepResult(this.id(), DataHubUpgradeState.FAILED);
+      }
+    };
+  }
+
+  /**
+   * Builds connector configuration by merging database connection details, Kafka settings, and
+   * explicit configuration overrides in that priority order.
+   */
+  @com.google.common.annotations.VisibleForTesting
+  Map<String, Object> buildConnectorConfiguration() {
+    Map<String, Object> config = new HashMap<>();
+
+    injectDatabaseConnection(config);
+    injectKafkaConnection(config);
+
+    if (debeziumConfig.getConfig() != null) {
+      config.putAll(debeziumConfig.getConfig());
+    }
+
+    try {
+      log.info(
+          "Built connector configuration with {} properties",
+          opContext.getObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(config));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
```

Comment:
Is it worth throwing a runtime exception for this log message? Feels like a warn would be more appropriate. If this intended to catch a failure scenario later on would be better to have it in validation flow rather than just using a log message to validate serialization to string.